distribution/client/src/lib/update.ts

```diff
@@ -163,57 +191,70 @@ export default class Update {
        while it's not been yet marked as tainted in the backend?
      * how to report that borked case in a clear way
   */
-  restartJenkins(rollingBack?: boolean) { // Add param to stop recursion?
-    Supervisord.restartProcess('jenkins');
-
-    const messageWhileRestarting = 'Jenkins should now be online, health checking!';
-    UI.publish(messageWhileRestarting);
-    logger.info(messageWhileRestarting);
-
-    // FIXME: actually now I'm thinking throwing in HealthChecker might provide a more
-    // consistent promise usage UX here.
-    // checking healthState.health value is possibly a bit convoluted (?)
-    this.healthChecker.check()
-      .then( healthState => {
-        if (healthState.healthy) {
-          logger.info('Jenkins healthcheck after restart succeeded! Yey.');
-        } else {
-
-          // if things are wrong twice, stop trying and just holler for help
-          if (rollingBack) {
-
-            // Quick notice sketch, but I do think we need a very complete and informative message
-            const failedToRollbackMessage =
-              'Ooh noes :-(. We are terribly sorry but it looks like Jenkins failed to ' +
-              'upgrade, but even after the automated rollback we were unable to bring ' +
-              'to life. Please report this issue to the Jenkins Evergreen team. ' +
-              'Do not shutdown your instance as we have been notified of this failure ' +
-              'and are trying to understand what went wrong to push a new update that ' +
-              'will fix things.';
-            logger.error(failedToRollbackMessage);
-            UI.publish(failedToRollbackMessage);
-
-            // Not throwing an Error here as we want the client to keep running and ready
-            // since the next available UL _might_ fix the issue
-          } else {
-
-            const errorMessage = `Jenkins detected as unhealthy. Rolling back to previous update level (${healthState.message}).`;
-            UI.publish(errorMessage);
-            logger.warn(errorMessage);
-
-            this.snapshotter.revertToLevelBefore(this.getCurrentLevel());
-            this.revertToPreviousUpdateLevel();
-            this.restartJenkins(true);
-          }
-        }
+  restartJenkins() {
+
+    const jenkinsIsRestarting = Supervisord.restartProcess('jenkins');
+
```

Comment:
What is with all these excessive `\n\n`?

Don't bother fixing in this monolithic pull request, but kindly fix your editor, my 13" notebook only has so much vertical space :smiling_imp:

Reply:
It's manually hard-wrapped for readability. So nothing I can fix on the editor side :-).

More seriously, I could _probably_ remove one or two newlines, but looking at the whole `restartJenkins` method, it does not look like there's that much unnecessary newlines there.
I guess I can remove L197 and L209, but the rest looks mostly fine to me.

![image](https://user-images.githubusercontent.com/223853/47009786-a0748080-d13d-11e8-8fa3-1dc664181d57.png)

ah #opinions :-)

Reply:
fixed with last push